### DISCUSSION TITLE
Misplaced sql filter inverting in sqlalchemy datastore?

### DISCUSSION THREAD

**Discussion by vadympop** [agronholm/apscheduler#858] [General]
![image](https://github.com/agronholm/apscheduler/assets/67959852/b161586a-c936-4e3b-a555-6928be2bae97)

I think this filter inverting is misplaced or can you explain why it's here?

**Comment by agronholm** [comment]
It sure looks that way at a glance. I need to check the associated tests to figure out why they're not failing then.

**Reply by agronholm** [reply]
That wasn't the only problem with the cleanup procedure: the MongoDB data store wasn't publishing the appropriate events.

**Reply by agronholm** [reply]
I don't understand. There are fairly extensive tests to make sure this works right, yet I'm not seeing any event being dispatched by the MongoDB store. I need to examine this more closely once I'm better rested.

**Reply by vadympop** [reply]
Okay, thank you
